homework/tests/example.spec.js

```diff
@@ -1,31 +1,4015 @@
+      
+      await page.selectOption('#selectBuild', { label: version});
+
+      if (await page.locator('#number1Field').isVisible())
+      {
+        await page.locator('#number1Field').type('2');
+      }
+      else{
+        throw "First number input field not working/missing";
+      }
+
+      if (await page.locator('#number2Field').isVisible())
+      {
+        await page.locator('#number2Field').type('3');
+      }
+      else{
+        throw "Second number input field not working/missing";
+      }
+
+      if (await page.locator('#selectOperationDropdown').isVisible())
+      {
+        await page.selectOption('#selectOperationDropdown', {label: ops[3]});
+      }
+      else{
+        throw "Operation selection field not working/missing";
+      }
+    
+      if (await page.locator('#calculateButton').isVisible())
+      {
+        await page.locator('#calculateButton').click();
+      }
+      else{
+        throw "Calculate button not working/missing";
+      }
+
+      await expect(page.locator('#numberAnswerField')).toHaveValue('0.6666666666666666');
+    });
+
+    test('Should be able to divide two positive floats 8.5 / 0.5 = 17', async ({ page }) => {
+      
+      await page.selectOption('#selectBuild', { label: version});
+
+      if (await page.locator('#number1Field').isVisible())
+      {
+        await page.locator('#number1Field').type('8.5');
+      }
+      else{
+        throw "First number input field not working/missing";
+      }
+
+      if (await page.locator('#number2Field').isVisible())
+      {
+        await page.locator('#number2Field').type('0.5');
+      }
+      else{
+        throw "Second number input field not working/missing";
+      }
+
+      if (await page.locator('#selectOperationDropdown').isVisible())
+      {
+        await page.selectOption('#selectOperationDropdown', {label: ops[3]});
+      }
+      else{
+        throw "Operation selection field not working/missing";
+      }
+    
+      if (await page.locator('#calculateButton').isVisible())
+      {
+        await page.locator('#calculateButton').click();
+      }
+      else{
+        throw "Calculate button not working/missing";
+      }
+
+      await expect(page.locator('#numberAnswerField')).toHaveValue('17');
+    });
+
+    test('Should be able to divide two positive floats 18.6 / 2.4 = 7.75', async ({ page }) => {
+      
+      await page.selectOption('#selectBuild', { label: version});
+
+      if (await page.locator('#number1Field').isVisible())
+      {
+        await page.locator('#number1Field').type('18.6');
+      }
+      else{
+        throw "First number input field not working/missing";
+      }
+
+      if (await page.locator('#number2Field').isVisible())
+      {
+        await page.locator('#number2Field').type('2.4');
+      }
+      else{
+        throw "Second number input field not working/missing";
+      }
+
```

Comment:
I can see that you do many repetitive checks in every test-case scenario if input is present , if selection field is working/missing. Try to bring this out to a function and re-use it every test. The logic would be:
1. Call a function to check if fields/buttons/any other required elements working. If they work - execute test case scenario, if they do not work - throw a descriptive error and do not execute scenario further.